galaxy_ng/app/common/proxy_url.py

```diff
@@ -0,0 +1,74 @@
+"""
+This module is temporarily added to address the
+issue AAH-291.
+
+The change that this module is solving
+is going to be implemented on pulpcore
+and then this module and its references
+can be removed or replaces with pulpcore
+implementation.
+"""
+from typing import List, Tuple, Optional
+from urllib3.util import parse_url, Url
+
+
+def get_parsed_url(parsed: Url, auth_items: Optional[List[Optional[str]]]) -> str:
+    """
+    parsed: Url instance
+    auth_items: Optional list of str ["user": "password"]
+    returns: str e.g: http://user:pass@foo.bar:9999
+    """
+    return Url(
+        scheme=parsed.scheme,
+        auth=auth_items and ":".join(auth_items) or None,
+        host=parsed.host,
+        port=parsed.port,
+        path=parsed.path,
+        query=parsed.query,
+        fragment=parsed.fragment,
+    ).url
+
+
+def strip_auth_from_url(url: str) -> Tuple[str, Optional[str], Optional[str]]:
```

Comment:
Could also return a dict and join_proxy_url() could use a fstring or similar formatting to build the proxy_url.

Again, don't know if it matters if this is a temp shim.

Reply:
As I used the pattern `var = value if condition else other_value` it was easy to set the `other_value` as `(None, None, None)` instead of creating a whole new empty dict or more if conditions.